packages/block-directory/src/store/actions.js

```diff
@@ -146,3 +150,54 @@ export function removeInstalledBlockType( item ) {
 		item,
 	};
 }
+
+/**
+ * Returns an action object used to indicate install in progress
+ *
+ */
+export function fetchInstallBlock() {
+	return {
+		type: 'FETCH_INSTALL_BLOCK',
+		isInstalling: true,
+	};
+}
+
+/**
+ * Returns an action object used to indicate install has completed
+ *
+ */
+export function receiveInstallBlock() {
+	return {
+		type: 'RECEIVE_INSTALL_BLOCKS',
+		isInstalling: false,
+	};
+}
+
+/**
+ * Sets an error notice string to be displayed to the user
+ *
+ * @param {string} blockId The ID of the block plugin. eg: my-block
+ * @param {string} noticeId The ID of the message used to determine which notice to show.
+ *
+ */
+export function setErrorNotice( blockId, noticeId ) {
+	return {
+		type: 'SET_ERROR_NOTICE_ID',
+		blockId,
+		noticeId,
+	};
+}
+
+/**
+ * Sets the error noticeId to empty for specific block
+ *
+ * @param {string} blockId The ID of the block plugin. eg: my-block
+ *
+ */
+export function clearErrorNotice( blockId ) {
+	return {
+		type: 'SET_ERROR_NOTICE_ID',
+		blockId,
+		noticeId: '',
```

Comment:
Thinking we should dispatch `noticeId: null` here instead and have the reducer delete the key value pair when `noticeId === null`. That way, notices are actually removed from the reducer state.

Reply:
I've taken another spin on this issue. It appears it only happens with that specific block. The other blocks don't show the same delay. Its hard to get a good sense of why since we don't yet have a proper testing environment to identify the cause, nor do we have a large number of blocks to test with. It could be because of how the injected script executes, maybe the size of the script matters (its the largest bundle i tested with), it could also be related to the block itself. That block is actually broken (if you refresh the page, you'll notice it breaks the page... something related to media-views.js and context...).

In the end, I would suggest that since the delay is not introduced here, and has a small affect on the experience (doesn't reproduce with other blocks),  we should cycle back to address it in another PR. Thoughts?